pkg/proxy/envoyproxy.go

```diff
@@ -0,0 +1,88 @@
+// Copyright 2016-2017 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package proxy
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/cilium/cilium/pkg/envoy"
+	"github.com/cilium/cilium/pkg/policy"
+
+	"github.com/spf13/viper"
+)
+
+// the global Envoy instance
+var envoyProxy *envoy.Envoy
+
+// EnvoyRedirect implements the Redirect interface for a l7 proxy
+type EnvoyRedirect struct {
+	id     string
```

Comment:
It is the redirect's identifier (a parameter given to proxy.CreateOrUpdateRedirect()), which we use when building the identifiers in Envoy configuration so that we can map between Envoy logs and Cilium logs if need be.